### PR TITLE
chore: Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by masone**
includes a useful recommendation for the first time ;)
https://github.com/smg-automotive/api-client-pkg/security/code-scanning/1

### CODE REVIEW

File: src/pathParameters.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const replaceParameters = ({
   path: string;
   params?: Record<string, string | number>;
 }) => {
-  const parameters = path.match(/{(.*?)}/g) || [];
+  const parameters = path.match(/{([A-Za-z0-9_-]+)}/g) || [];
```
Comment (masone) [inline src/pathParameters.ts:20]: used to replace placeholders in urls, eg. `/foo/{paramName}`

Comment (masone) [reply src/pathParameters.ts:20]: https://regex101.com/r/p978Zs/1

Comment (Averethel) [inline src/pathParameters.ts:20]: How is this related to adding code ql?

Comment (masone) [reply src/pathParameters.ts:20]: https://github.com/smg-automotive/api-client-pkg/security/code-scanning/1 warning from the security scanner

> Some regular expressions take a long time to match certain input strings to the point where the time it takes to match a string of length n is proportional to nk or even 2n. Such regular expressions can negatively affect performance, or even allow a malicious user to perform a Denial of Service ("DoS") attack by crafting an expensive input string for the regular expression to match.

if the parameter would be`{{{{{{{{{{{{{{{{{{` (and more), the match would end up in a loop because `{.*` would recursively match

**Review by Averethel — APPROVED** [review]
